test/saucelabs/test.py

```diff
@@ -155,35 +128,28 @@ def parse_args(args):
     caps_used = [top_browser] if singlebrowser else desired_cap_list
 
     for current_desired_cap in caps_used:
-        print "+--> Start test with " + current_desired_cap['platform'] + \
-            " " + current_desired_cap['browserName'] + " (" + current_desired_cap['version'] + ")"
-        driver = webdriver.Remote(
-            command_executor='http://' +
-            saucelabs_user +
-            ':' +
-            saucelabs_key +
-            '@ondemand.saucelabs.com:80/wd/hub',
-            desired_capabilities=current_desired_cap)
-        driver.implicitly_wait(DEFAULT_WAIT_FOUND)
-        if driver.name == "MicrosoftEdge":
-            # print 'Force set version, strange...'
-            driver.desired_capabilities['version'] = current_desired_cap['version']
         try:
-            if driver.name == "internet explorer" or driver.name == "opera" or driver.name == "safari":
-                # Use specific test list for IE, Opera and Safari
-                DoTestCurrentrowser = doTestsLight
-            elif driver.name == "MicrosoftEdge":
-                # Use full test list for Edge
-                DoTestCurrentrowser = doTestsEdge
-            else:
-                # Use full test list for all browser
-                DoTestCurrentrowser = doTests
-            for k, dotest in DoTestCurrentrowser.iteritems():
-                if k in tests or len(tests) == 0:
+            print "+--> Start test with " + current_desired_cap['platform'] + \
+                " " + current_desired_cap['browserName'] + " (" + current_desired_cap['version'] + ")"
+            driver = webdriver.Remote(
+                command_executor='http://' +
+                saucelabs_user +
+                ':' +
+                saucelabs_key +
+                '@ondemand.saucelabs.com:80/wd/hub',
+                desired_capabilities=current_desired_cap)
+            driver.implicitly_wait(DEFAULT_WAIT_FOUND)
+
+            if driver.name == "MicrosoftEdge":
+                print 'Force set version, strange...'
```

Comment:
I don't know why but with Edge you loose the 'version' information. 
So i force to rewrite it, not so nice, i'm agree, if you have an other solution ?